syntax = "proto3";

import "player.proto";

enum PBCombatActions 
{
    CA_NONE            = 0;
    CA_ATTACK          = 1;
    CA_DODGE           = 2;
    CA_BLOCK           = 3;
    CA_SURRENDER       = 4;
}

message PBCombat 
{
    int64 starter_user_id               = 1;
    int64 opponent_user_id              = 2;
    int32 wager                         = 3;
    int32 turn                          = 4;
    PBCombatActions starter_action      = 5;
    PBCombatActions opponent_action     = 6;
    int32 starter_player_id             = 7;
    int32 opponent_player_id            = 8;
    PBPlayer starter_user_info          = 9;
    PBPlayer opponent_user_info         = 10;
}